src/plugins/intel_cpu/src/cpu_streams_calculation.cpp

```diff
@@ -40,43 +41,53 @@ std::vector<std::vector<int>> get_streams_info_table(const int input_streams,
-        int numa_node_id = stream_info[STREAM_NUMA_NODE_ID];
         int socket_id = stream_info[STREAM_SOCKET_ID];
         int node_start = one_proc_table.size() == 1 ? 0 : 1;
         int node_end = one_proc_table.size() == 1 ? 1 : one_proc_table.size();
-        for (int n = MAIN_CORE_PROC; n <= HYPER_THREADING_PROC; n++) {
-            for (int index = node_start; index < node_end; index++) {
-                if (((numa_node_id < 0) || (numa_node_id == one_proc_table[index][PROC_NUMA_NODE_ID])) &&
-                    ((socket_id < 0) || (socket_id == one_proc_table[index][PROC_SOCKET_ID]))) {
-                    if (0 != one_proc_table[index][n]) {
-                        stream_info[PROC_TYPE] = n;
-                        if (total_threads <= one_proc_table[index][n]) {
-                            stream_info[THREADS_PER_STREAM] = total_threads;
-                            stream_info[STREAM_NUMA_NODE_ID] = one_proc_table[index][PROC_NUMA_NODE_ID];
-                            stream_info[STREAM_SOCKET_ID] = one_proc_table[index][PROC_SOCKET_ID];
-                            streams_info_table.push_back(stream_info);
-                            return;
-                        } else {
-                            stream_info[THREADS_PER_STREAM] = one_proc_table[index][n];
-                            stream_info[STREAM_NUMA_NODE_ID] = one_proc_table[index][PROC_NUMA_NODE_ID];
-                            stream_info[STREAM_SOCKET_ID] = one_proc_table[index][PROC_SOCKET_ID];
-                            streams_info_table.push_back(stream_info);
-                            total_threads -= one_proc_table[index][n];
+        // When n_mode is 3, the following loop only selects CPUs on socket with the same id as current_socket_id.
+        // When n_mode is 2, the following loop only selects CPUs on sockets with id different from current_socket_id.
+        // When n_mode is 1, the following loop selects CPUs on all sockets.
+        for (int n_mode = current_socket_id < 0 ? 1 : 3; (n_mode > 0) && (total_threads > 0); n_mode--) {
+            for (int n = MAIN_CORE_PROC; (n <= HYPER_THREADING_PROC) && (total_threads > 0); n++) {
+                for (int index = node_start; (index < node_end) && (total_threads > 0); index++) {
+                    if (((n_mode == 1) && ((socket_id < 0) || (socket_id == one_proc_table[index][PROC_SOCKET_ID]))) ||
+                        ((n_mode == 2) && (current_socket_id != one_proc_table[index][PROC_SOCKET_ID]) &&
+                         ((socket_id < 0) || (socket_id == one_proc_table[index][PROC_SOCKET_ID]))) ||
+                        ((n_mode == 3) && (current_socket_id == one_proc_table[index][PROC_SOCKET_ID]) &&
+                         ((socket_id < 0) || (socket_id == one_proc_table[index][PROC_SOCKET_ID])))) {
+                        if ((0 != one_proc_table[index][n]) && ((ALL_PROC == target_proc) || (n == target_proc))) {
```

Comment:
This part of code is totaly unreadable. Another one point that we need high-level interface to work with the table (like `table.get_socket_id(index)`). Also it is better to avoid possibility to set negative values for things like `socket_id`, otherwise you have to check it dozens of times.
Not necessary to fix in this PR. Just let you know.

Reply:
Got it. Will refine the code for using high level interface in flow in following PRs.

Using `-1` because here need a invalid initial value when `0` is valid value. @dmitry-gorokhov I was wondering if you think the maximum value of int is better than `-1` here?

Reply:
Better to avoid invalid numbers at all. Anyway lets discuss refactoring separatelly.